### PR TITLE
#165: integrate itde 1.0.0 and add useful return values to api functions

### LINKED ISSUE #165: API should return useful information
# Background

- Currently, the API prints at most to system out, but doesn't return useful return values

# Acceptance Criteria

- Export API returns the path to the exported container file
- Build API returns the build images
- GenerateLanguageActivation API returns the LanguageActivation
- ...

### PR DESCRIPTION
**Pull request by tomuben**
- added return values to API functions
- implemented test container injection
- added tests for export and push tests

fixes #165

### CODE REVIEW

File: exasol_script_languages_container_tool/lib/api/generate_language_activation.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 def generate_language_activation(
         flavor_path: str,
         bucketfs_name: str,
         bucket_name: str,
         container_name: str,
-        path_in_bucket: str = ''):
+        path_in_bucket: str = '') -> Tuple[str, str, str]:
     """
     Generate the language activation statement.
+    :return: A tuple of language definition statements: The first one is the alter session statement,
```
Comment (tkilias) [inline exasol_script_languages_container_tool/lib/api/generate_language_activation.py:19]: I think, at some point, we should return the alias and url separated for each aliases. Or, an object which stores them and can generate the alter session and system command. With that, we could also change the alias.

**Review by Nicoretti — APPROVED** [review]

File: test/test_docker_pull.py [hunk 2]
```diff
@@ -11,23 +11,23 @@ class DockerPullTest(unittest.TestCase):
     def setUp(self):
         print(f"SetUp {self.__class__.__name__}")
         self.test_environment = exaslct_utils.ExaslctTestEnvironmentWithCleanUp(self, exaslct_utils.EXASLCT_DEFAULT_BIN)
-        self.registry_container, self.registry_host, self.registry_port = self.test_environment.create_registry()
-        self.test_environment.clean_images()
```
Comment (tkilias) [inline test/test_docker_pull.py:15]: This clean you should keep, to cleanup potential still existing images in the cache of the docker service

**Review by tkilias — CHANGES_REQUESTED** [review]

File: exasol_script_languages_container_tool/cli/commands/build.py [hunk 3]
```diff
@@ -57,7 +56,7 @@ def build(flavor_path: Tuple[str, ...],
     If stages are cached in a docker registry, they command is going to pull them,
```
[the remark this answers is not in the document]
Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/build.py:56]: Fixed

File: exasol_script_languages_container_tool/cli/commands/run_db_tests.py [hunk 4]
```diff
@@ -70,6 +70,9 @@
 @click.option('--reuse-test-environment/--no-reuse-test-environment', default=False,
               help="Reuse the whole test environment with docker network, test container, "
                    "database, database setup and uploaded container")
+@click.option('--test-container-folder', type=click.Path(exists=True, file_okay=False, dir_okay=True),
+              default="./test_container",
```
[the remark this answers is not in the document]
Comment (tomuben) [reply exasol_script_languages_container_tool/cli/commands/run_db_tests.py:74]: You mean rename $root/test -> $root/tests?

File: exasol_script_languages_container_tool/lib/api/generate_language_activation.py (hunk 1 above)
Comment (tomuben) [reply exasol_script_languages_container_tool/lib/api/generate_language_activation.py:19]: Created #171

File: test/test_docker_pull.py (hunk 2 above)
Comment (tomuben) [reply test/test_docker_pull.py:15]: The clean is still there, check line 18

File: exasol_script_languages_container_tool/cli/commands/run_db_tests.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_script_languages_container_tool/cli/commands/run_db_tests.py:74]: I mean, that you renamed ./tests -> to ./test_container and as I said probably a good idea.

**Review by tkilias — APPROVED** [review]

**Pull request merged**